build_tools/cmake/iree_check_test.cmake

```diff
@@ -121,6 +121,7 @@ function(iree_check_test)
         TEST_EXECUTABLE=$<TARGET_FILE:iree_modules_check_iree-check-module>
     )
     set_property(TEST ${_TEST_NAME} PROPERTY ENVIRONMENT ${_ENVIRONMENT_VARS})
+    iree_add_test_environment_properties(${_TEST_NAME})
```

Comment:
The check tests don't use these environment variables. We can leave them for consistency I guess.

Reply:
> > Haven't looked into this; but this is awesome, thanks Scott for pushing forward on this! Can I ask that we manually invoke some of the tests only run on post submit before landing this? This is non-trivial changes to the build system. :)
> 
> Swiftshader tests now run on presubmit. Do you want me to manually trigger Turing and Android tests?

This is also tricky to get test coverage for, since ideally it would be a no-op on the CI as configured today. Going forward, local development builds will be free to disable components that they care about, but we won't have automated coverage of the entire matrix of configurations.